workflow-manager/task/task.go

```diff
@@ -0,0 +1,156 @@
+// workers by workflow-manager. A _task_ is a work item at the data share
+// processor application layer (i.e., intake of a batch or aggregation of many
+// batches); its name is chosen to distinguish from Kubernetes-level _jobs_.
+package task
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"sync"
+
+	"cloud.google.com/go/pubsub"
+)
+
+// Task is a task that can be enqueued into an Enqueuer
+type Task interface {
+	// MarshalToJson marshals self into a JSON message
+	MarshalToJson() ([]byte, error)
+}
+
+// Aggregation represents an aggregation task
+type Aggregation struct {
+	// AggregationID is the identifier for the aggregation
+	AggregationID string `json:"aggregation-id"`
+	// AggregationStart is the start of the range of time covered by the
+	// aggregation in UTC, with minute precision, formatted like
+	// "2006/01/02/15/04"
+	AggregationStart string `json:"aggregation-start"`
+	// AggregationEnd is the end of the range of time covered by the aggregation
+	// in UTC, with minute precision, formatted like "2006/01/02/15/04"
+	AggregationEnd string `json:"aggregation-end"`
+	// Batches is the list of batch ID date pairs of the batches aggregated by
+	// this task
+	Batches []Batch `json:"batches"`
+}
+
+func (a Aggregation) MarshalToJson() ([]byte, error) {
+	return json.Marshal(a)
+}
+
+// Batch represents a batch included in an aggregation task
+type Batch struct {
+	// ID is the batch ID. Typically a UUID.
+	ID string `json:"id"`
+	// Time is the timestamp on the batch, in UTC, with minute precision,
+	// formatted like "2006/01/02/15/04"
+	Time string `json:"time"`
+}
+
+type IntakeBatch struct {
+	// AggregationID is the identifier for the aggregaton
+	AggregationID string `json:"aggregation-id"`
+	// BatchID is the identifier of the batch. Typically a UUID.
+	BatchID string `json:"batch-id"`
+	// Time is the timestamp on the batch, in UTC, with minute precision,
+	// formatted like "2006/01/02/15/04"
+	Date string `json:"date"`
+}
+
+func (i IntakeBatch) MarshalToJson() ([]byte, error) {
+	return json.Marshal(i)
+}
+
+// Enqueuer allows enqueuing tasks.
+type Enqueuer interface {
+	// Enqueue enqueues a task to be executed later. The provided completion
+	// function will be invoked once the task is either successfully enqueued or
+	// some unretryable error has occurred. A call to Stop() will not return
+	// until completion functions passed to any and all calls to Enqueue() have
+	// returned.
+	Enqueue(task Task, completion func(error))
+	// Stop blocks until all tasks passed to Enqueue() have been enqueued in the
+	// underlying system, and all completion functions pased to Enqueue() have
+	// returned, and so it is safe to exit the program without losing any tasks.
+	Stop()
+}
+
+// CreatePubSubTopic creates a PubSub topic with the provided ID. Returns error
+// on failure.
+func CreatePubSubTopic(ctx context.Context, project string, topicID string) error {
```

Comment:
I'd create the topics with Terraform and not in the workflow-manager.

Reply:
That's the plan -- this exists only to support local development cases, working with the PubSub emulator, since it is not going to be managed by Terraform and you can't use `gcloud` to create topics or subscriptions.

Reply:
Remove ctx from function call and create a separate context here:

```
ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
defer cancel()
```